plugin-infra/go-plugin-access/src/main/java/com/thoughtworks/go/plugin/access/configrepo/ConfigRepoPluginInfoBuilder.java

```diff
@@ -35,11 +36,14 @@ public ConfigRepoPluginInfoBuilder(ConfigRepoExtension extension) {
     public ConfigRepoPluginInfo pluginInfoFor(GoPluginDescriptor descriptor) {
         PluggableInstanceSettings pluggableInstanceSettings = getPluginSettingsAndView(descriptor, extension);
 
-        return new ConfigRepoPluginInfo(descriptor, image(descriptor.id()), pluggableInstanceSettings);
+        return new ConfigRepoPluginInfo(descriptor, image(descriptor.id()), pluggableInstanceSettings, capabilities(descriptor.id()));
     }
 
     private com.thoughtworks.go.plugin.domain.common.Image image(String pluginId) {
         return extension.getIcon(pluginId);
     }
 
+    private Capabilities capabilities(String pluginId) {
+        return extension.getCapabilities(pluginId);
+    }
```

Comment:
👍 

The intention behind this is to cache the calls made to the plugin, that always return the same data. Capabilities, get view, get icon, etc. I noticed a few days ago that `GoConfigPluginService` directly calls `getCapabilities` on the extension, instead of looking up the metadata store. https://github.com/gocd/gocd/blob/84ae8699545ee85320fbaeb0702acc4009d74032/server/src/main/java/com/thoughtworks/go/config/GoConfigPluginService.java#L62-L64

Would you mind changing that reference to use the metadata store instead (in a separate commit)?

Reply:
Of course. Good catch